app/uk/gov/hmrc/agentepayeregistration/validators/AgentEpayeRegistrationValidator.scala

```diff
@@ -113,4 +132,32 @@ object AgentEpayeRegistrationValidator {
       Valid(())
     else
       Invalid(Failure("INVALID_FIELD", s"The $propertyName field is not a valid phone number"))
+
+  def parseISODate(dateTxt: String) = LocalDate.parse(dateTxt, ISODateTimeFormat.date())
+
+  private[validators] def isISODate(param: String)(paramName: String) =
+    Try(parseISODate(param)) match {
+    case Success(_) =>
+      Valid(())
+    case _ =>
+      Invalid(Failure("INVALID_DATE_FORMAT", s"'$paramName' date must be in ISO format (yyyy-MM-dd)"))
+  }
+
+  private[validators] def isInPast(date: LocalDate)(paramName: String) = {
+    val today = LocalDate.now(DateTimeZone.UTC)
+    if (today.isAfter(date))
+      Valid(())
+    else
+      Invalid(Failure("INVALID_DATE_RANGE", s"'$paramName' date must be in the past"))
+  }
+
+  private[validators] def isValidDateRange(dateFrom: LocalDate, dateTo: LocalDate) = {
+    if (dateFrom.isAfter(dateTo))
```

Comment:
`isAfter` may throw an exception in some cases. Shouldn't we consider using Try here too?